### PR TITLE
📖 Update root README file

### PR DESCRIPTION
**Pull request by furkatgofurov7**
**What this PR does / why we need it**:
Updates root readme file with latest changes since release v1.1.0 is out and CAPI release v1.1.2 is being used as of now. 
Also, updates a bit deployment process workflow

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -60,20 +67,20 @@ provider:
 1. Install Ironic. There are a couple of ways to do it.
     - Run within a Kubernetes cluster as a pod, refer to [deploy.sh](https://github.com/metal3-io/baremetal-operator/blob/main/tools/deploy.sh)
       script.
-    - Outside of a Kubernetes cluster as a container. Please refer to [run_local_ironic.sh](https://github.com/metal3-io/baremetal-operator/blob/main/tools/run_local_ironic.sh).
+    - Outside of a Kubernetes cluster as a container. Please refer to [run_local_ironic.sh](https://github.com/metal3-io/baremetal-operator/blob/main/tools/run_local_ironic.sh) script.
```
Comment (lentzi90) [inline README.md:70]: nit
```suggestion
    - Outside of a Kubernetes cluster as a container. Please refer to the [run_local_ironic.sh](https://github.com/metal3-io/baremetal-operator/blob/main/tools/run_local_ironic.sh) script.
```

File: README.md [hunk 2]
```diff
@@ -60,20 +67,20 @@ provider:
...
 
 ## Pivoting ⚠️
 
-Starting from `v0.5.0` release of Cluster-api-provider-metal3, Baremetal Operator is decoupled
-from Cluster-api-provider-metal3 deployments when deployed via `clusterctl`. For that reason,
+Starting from `v0.5.0` release of Cluster API Provider Metal3, Baremetal Operator is decoupled
+from Cluster API Provider Metal3 deployments when deployed via `clusterctl`. For that reason,
 when performing `clusterctl move`, custom objects outside of the Cluster API chain or not part
-of CAPM3 will not be pivoted to a target cluster. Example to those objects is BareMetalHost, its
+of CAPM3 will not be pivoted to a target cluster. An example of those objects is BareMetalHost, it's
```
Comment (lentzi90) [inline README.md:79]: I think it should be "its" here, but maybe we could change it a bit to clarify this is not just about BMO resources?
```suggestion
of CAPM3 will not be pivoted to a target cluster. An example of those objects is BareMetalHost, or user created ConfigMaps and Secrets.
```

File: README.md [hunk 3]
```diff
@@ -19,35 +19,42 @@ cluster on top of bare metal infrastructure using Metal3.
...
 
 ## Deploying the metal3 provider
 
 The recommended method is using
 [Clusterctl](https://main.cluster-api.sigs.k8s.io/clusterctl/overview.html).
 
-Starting from `v0.5.0` release of Cluster-api-provider-metal3, Baremetal Operator is decoupled
-from Cluster-api-provider-metal3 deployments when deployed via `clusterctl`. For that reason,
+Starting from `v0.5.0` release of Cluster API Provider Metal3, Baremetal Operator is decoupled
+from Cluster API Provider Metal3 deployments when deployed via `clusterctl`. For that reason,
```
Comment (namnx228) [inline README.md:30]: just small nit. Change is optional.
```suggestion
from Cluster API Provider Metal3 deployments when deployed via `clusterctl`. For this reason,
```

File: README.md [hunk 4]
```diff
@@ -19,35 +19,42 @@ cluster on top of bare metal infrastructure using Metal3.
...
 Baremetal Operator will not be installed when initializing the Metal3 provider with clusterctl,
 and its CRDs and controller need to be manually installed. Example flow of installing Metal3
 provider:
 
 1. Install Cluster API core, bootstrap and control-plane providers. This will also install
-  cert-manager if it is not already installed.
+  a cert-manager if it is not already installed. To have more verbose logs you can use the -v flag
```
Comment (namnx228) [inline README.md:36]: Better if it is 
```suggestion
  cert-manager if it is not already installed. To have more verbose logs you can use the -v flag
```

**Review by namnx228 — COMMENTED** [review]
LGTM. just some nits

**Review by fmuyassarov — APPROVED** [review]
Agree with below comments

File: README.md (hunk 2 above)
Comment (furkatgofurov7) [reply README.md]: Thanks, fixed.

File: README.md (hunk 4 above)
Comment (furkatgofurov7) [reply README.md]: Thanks, fixed.

**Comment by furkatgofurov7** [comment]
Thanks for the reviews all! I have addressed all of them, take a look again.

**Review by Rozzii — APPROVED** [review]

**Review by lentzi90 — APPROVED** [review]

**Review by namnx228 — COMMENTED** [review]
/lgtm

**Pull request merged**